### ISSUE TITLE
Not able to install pyodbc on linux Red Hat

### ISSUE THREAD

**Issue by msinghnatt** [mkleehammer/pyodbc#398]
Please first make sure you have looked at:

* Documentation: https://github.com/mkleehammer/pyodbc/wiki
* Other issues

### Environment

To diagnose, we usually need to know the following, including version numbers.  On Windows, be
sure to specify 32-bit Python or 64-bit:

- Python:    **Python 3.6.5** (default, May 21 2018) [GCC 4.4.7 20120313 (Red Hat 4.4.7-18)] on linux
- pyodbc: 4.0.23
- OS:  Red Hat Enterprise Linux Server release 6.9 (Santiago) 64 bit
- DB:   
- driver:  No MS ODBC driver installed on Linux server yet.

### Issue
Trying to install pyodbc on Linux server(Red Hat). Can't use pip as I don't have internet connection on server. So download the source and used **python3 setup.py install** command to install manually.

Please Note, **No MS ODBC driver installed on Linux server yet**.  Is there a dependency on MS ODBC??

Below is the summary of logs  with errors.

running install
running bdist_egg
running egg_info
writing pyodbc.egg-info/PKG-INFO
writing dependency_links to pyodbc.egg-info/dependency_links.txt
writing top-level names to pyodbc.egg-info/top_level.txt
reading manifest file 'pyodbc.egg-info/SOURCES.txt'
reading manifest template 'MANIFEST.in'
writing manifest file 'pyodbc.egg-info/SOURCES.txt'
installing library code to build/bdist.linux-x86_64/egg
running install_lib
running build_ext
building 'pyodbc' extension
gcc -pthread -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -fPIC -DPYODBC_VERSION=4.0.23 -I/usr/local/include/python3.6m -c src/getdata.cpp -o build/temp.linux-x86_64-3.6/src/getdata.o -Wno-write-strings -DHAVE_UNISTD_H -DHAVE_PWD_H -DHAVE_SYS_TYPES_H -DHAVE_LONG_LONG -DSIZEOF_LONG_INT=8 -I/usr/include
cc1plus: warning: command line option "-Wstrict-prototypes" is valid for Ada/C/ObjC but not for C++
In file included from src/getdata.cpp:5:
src/pyodbc.h:56:17: error: sql.h: No such file or directory
src/pyodbc.h:57:20: error: sqlext.h: No such file or directory
In file included from src/getdata.cpp:5:
src/pyodbc.h:73: error: ‘IsSet’ declared as an ‘inline’ variable
src/pyodbc.h:73: error: ‘DWORD’ was not declared in this scope
src/pyodbc.h:73: error: ‘DWORD’ was not declared in this scope
src/pyodbc.h:73: error: initializer expression list treated as compound expression
src/pyodbc.h:74: error: expected ‘,’ or ‘;’ before ‘{’ token
src/pyodbc.h:140: error: ‘SQLSMALLINT’ was not declared in this scope
src/pyodbc.h:141: error: ‘SQLSMALLINT’ was not declared in this scope
In file included from src/getdata.cpp:7:
src/textenc.h:52: error: ‘SQLSMALLINT’ does not name a type
In file included from src/getdata.cpp:8:
src/pyodbcmodule.h:45: error: ‘HENV’ does not name a type
In file included from src/getdata.cpp:9:
src/cursor.h:20: error: ‘SQLSMALLINT’ does not name a type
src/cursor.h:28: error: ‘SQLULEN’ does not name a type
src/cursor.h:39: error: ‘SQLSMALLINT’ does not name a type
src/cursor.h:40: error: ‘SQLSMALLINT’ does not name a type
src/cursor.h:41: error: ‘SQLULEN’ does not name a type
src/cursor.h:42: error: ‘SQLSMALLINT’ does not name a type
src/cursor.h:46: error: ‘SQLPOINTER’ does not name a type
src/cursor.h:48: error: ‘SQLLEN’ does not name a type
src/cursor.h:49: error: ‘SQLLEN’ does not name a type
src/cursor.h:62: error: ‘SQLLEN’ does not name a type
src/cursor.h:74: error: ‘TIMESTAMP_STRUCT’ does not name a type
src/cursor.h:75: error: ‘DATE_STRUCT’ does not name a type
src/cursor.h:76: error: ‘TIME_STRUCT’ does not name a type
src/cursor.h:88: error: ‘HSTMT’ does not name a type
src/cursor.h:105: error: ISO C++ forbids declaration of ‘SQLSMALLINT’ with no type
src/cursor.h:105: error: expected ‘;’ before ‘*’ token
In file included from src/getdata.cpp:10:
src/connection.h:26: error: ‘HDBC’ does not name a type
src/connection.h:75: error: ‘SQLLEN’ does not name a type
src/connection.h:97: error: ISO C++ forbids declaration of ‘SQLSMALLINT’ with no type
src/connection.h:97: error: expected ‘;’ before ‘*’ token
src/connection.h:114: error: ‘SQLSMALLINT’ has not been declared
In file included from src/getdata.cpp:11:
src/errors.h:15: error: ‘HDBC’ has not been declared
src/errors.h:15: error: ‘HSTMT’ has not been declared
src/errors.h:43: error: ‘HDBC’ has not been declared
src/errors.h:43: error: ‘HSTMT’ has not been declared
src/errors.h:56: error: ‘bool HasSqlState’ redeclared as different kind of symbol
src/errors.h:50: error: previous declaration of ‘bool HasSqlState(PyObject*, const char*)’
src/errors.h:56: error: ‘HSTMT’ was not declared in this scope
src/errors.h:56: error: expected primary-expression before ‘const’
In file included from src/getdata.cpp:12:
src/dbspecific.h:20: error: ‘SQLUSMALLINT’ does not name a type
src/dbspecific.h:21: error: ‘SQLUSMALLINT’ does not name a type
src/dbspecific.h:22: error: ‘SQLUSMALLINT’ does not name a type
src/dbspecific.h:23: error: ‘SQLUINTEGER’ does not name a type
src/dbspecific.h:35: error: ‘DWORD’ does not name a type
src/dbspecific.h:36: error: ‘WORD’ does not name a type
src/dbspecific.h:37: error: ‘WORD’ does not name a type
In file included from src/getdata.cpp:13:
src/sqlwchar.h:33: error: ‘SQLSMALLINT’ does not name a type
src/sqlwchar.h:36: error: ‘SQLSMALLINT’ has not been declared
src/sqlwchar.h:75: error: ‘SQLSMALLINT’ has not been declared
src/sqlwchar.h:79: error: ‘SQLSMALLINT’ has not been declared
src/sqlwchar.h: In member function ‘void SQLWChar::init(PyObject*, int, PyObject*, const char*)’:
src/sqlwchar.h:40: error: ‘ctype’ was not declared in this scope
src/sqlwchar.h: In member function ‘Py_ssize_t SQLWChar::charlen() const’:
src/sqlwchar.h:89: error: ‘ctype’ was not declared in this scope
src/sqlwchar.h:89: error: ‘SQL_C_WCHAR’ was not declared in this scope
src/getdata.cpp: At global scope:
src/getdata.cpp:41: error: ‘IsBinaryType’ declared as an ‘inline’ variable
src/getdata.cpp:41: error: ‘SQLSMALLINT’ was not declared in this scope
src/getdata.cpp:42: error: expected ‘,’ or ‘;’ before ‘{’ token
src/getdata.cpp:39: warning: ‘byte* ReallocOrFreeBuffer(byte*, Py_ssize_t)’ declared ‘static’ but never defined
error: command 'gcc' failed with exit status 1

**Comment by v-chojas** [comment]
`error: sql.h: No such file or directory`
You need to install a [driver manager](https://github.com/mkleehammer/pyodbc/wiki/Drivers-and-Driver-Managers). [unixODBC](http://www.unixodbc.org/) is recommended.

**Comment by msinghnatt** [comment]
My goal is to connect SQL server 2016 database on windows from Linux server using python.
Do I need both unixODBC and Microsoft ODBC Driver 17 ?
Or installing unixODBC is enough to install pyodbc and connect to SQL server database on windows server?

**Comment by v-makouz** [comment]
Yes, both unixODBC and the Driver are needed. Installing the driver will also install unixODBC
